source/compiler/sc5.c

```diff
@@ -131,7 +131,9 @@ static char *errmsg[] = {
 /*089*/  "state variables may not be initialized (symbol \"%s\")\n",
 /*090*/  "public functions may not return arrays (symbol \"%s\")\n",
 /*091*/  "ambiguous constant; tag override is required (symbol \"%s\")\n",
-/*092*/  "functions may not return arrays of unknown size (symbol \"%s\")\n"
+/*092*/  "functions may not return arrays of unknown size (symbol \"%s\")\n",
+/*093*/  "\"__addressof\" operator is invalid in preprocessor expressions\n",
+/*094*/  "\"__addressof\" operator is invalid on local non-static variables\n"
```

Comment:
Why should using `__addressof` on local variables cause an error?

Reply:
> It would be even more useful it could compute the address at run-time.

You can already do that with `__emit`.
```Pawn
new addr = __emit addr.u.pri var;
```

Reply:
IMO, `__addressof` is much neater.